From c33f55cbb85abbb71ee9651a12861f1367e4d23d Mon Sep 17 00:00:00 2001
From: Min RK <benjaminrk@gmail.com>
Date: Fri, 12 Aug 2022 11:23:51 +0200
Subject: [PATCH 1/2] init cmake_args with $CMAKE_ARGS in Python build

allows customization of the build

required for cross-compilation in conda packages
---
 python/setup.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/python/setup.py b/python/setup.py
index ab3c25915e..bad3c2d703 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -1,4 +1,5 @@
 import os
+import shlex
 import subprocess
 import sys
 import sysconfig
@@ -41,6 +42,7 @@ def run(self):
 
     def build_extension(self, ext):
         extdir = os.path.abspath(os.path.dirname(self.get_ext_fullpath(ext.name)))
+        cmake_args = shlex.split(os.environ.get("CMAKE_ARGS", ""))
         cmake_args = ['-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=' + extdir,
                       '-DPython3_EXECUTABLE=' + sys.executable,
                       f'-DPython3_LIBRARIES={sysconfig.get_config_var("LIBDEST")}',

From b97ca05746a72f96a58461b767c0870571d2e3f1 Mon Sep 17 00:00:00 2001
From: Min RK <benjaminrk@gmail.com>
Date: Tue, 16 Aug 2022 16:26:50 +0200
Subject: [PATCH 2/2] respect $CMAKE_BUILD_PARALLEL_LEVEL in python builds

keep the default of 3
---
 python/setup.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/python/setup.py b/python/setup.py
index bad3c2d703..ec100e4649 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -51,9 +51,12 @@ def build_extension(self, ext):
         cfg = 'Debug' if self.debug else 'Release'
         build_args = ['--config', cfg]
         cmake_args += ['-DCMAKE_BUILD_TYPE=' + cfg]
-        build_args += ['--', '-j3']
 
         env = os.environ.copy()
+        # default to 3 build threads
+        if "CMAKE_BUILD_PARALLEL_LEVEL" not in env:
+            env["CMAKE_BUILD_PARALLEL_LEVEL"] = "3"
+
         import pybind11
         env['pybind11_DIR'] = pybind11.get_cmake_dir()
         env['CXXFLAGS'] = '{} -DVERSION_INFO=\\"{}\\"'.format(env.get('CXXFLAGS', ''),
